module-services/service-cellular/DTMFCode.cpp

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2017-2022, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#include "DTMFCode.hpp"
+#include <Utils.hpp>
+#include <stdexcept>
+
+DTMFCode::DTMFCode(const char digitChar)
+{
+    if ((digitChar >= '0' && digitChar <= '9') || digitChar == '*' || digitChar == '#') {
+        this->digitChar = digitChar;
+    }
+    else {
+        this->digitChar = 0;
+        throw std::out_of_range("Invalid number!");
+    }
+}
+DTMFCode::DTMFCode(const char *digitStr)
+{
+    auto digitTmpString = std::string{digitStr};
+    try {
+        auto digit = std::stoi(digitTmpString);
+        if (digit > 9) {
+            throw std::out_of_range("Invalid number!");
+        }
+        digitChar = digit + '0';
+    }
+    catch (std::invalid_argument &e) {
```

Comment:
here actually I would check for `*` and `#` prior to stoi. 
Having in mind that:

_exception is meant to handle something unexpected, not to create if-else possibilities_

Reply:
fixed